include/ibamr/FESurfaceDistanceEvaluator.h

```diff
@@ -0,0 +1,249 @@
+//      this list of conditions and the following disclaimer.
+//
+//    * Redistributions in binary form must reproduce the above copyright
+//      notice, this list of conditions and the following disclaimer in the
+//      documentation and/or other materials provided with the distribution.
+//
+//    * Neither the name of The University of North Carolina nor the names of
+//      its contributors may be used to endorse or promote products derived from
+//      this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+
+/////////////////////// INCLUDE GUARD ////////////////////////////////////
+
+#ifndef included_FESurfaceDistanceEvaluator
+#define included_FESurfaceDistanceEvaluator
+
+///////////////////////////// INCLUDES ///////////////////////////////////
+#include <map>
+#include <vector>
+
+#include "PatchHierarchy.h"
+#include "ibamr/IBFEMethod.h"
+#include "ibtk/IndexUtilities.h"
+#include "ibtk/ibtk_utilities.h"
+#include "ibtk/libmesh_utilities.h"
+#include "libmesh/boundary_mesh.h"
+#include "tbox/Pointer.h"
+
+/////////////////////////////// CLASS DEFINITION /////////////////////////////
+
+namespace IBAMR
+{
+/*!
+ * \brief class FESurfaceDistanceEvaluator is a utility class which is used to identify
+ * which line (triangle) in 2D (3D) elements are intersecting which grid cells.
+ */
+class FESurfaceDistanceEvaluator
+{
+public:
+    static const double s_large_distance;
+
+    /*!
+     * The only constructor of this class.
+     */
+    FESurfaceDistanceEvaluator(const std::string& object_name,
+                               SAMRAI::tbox::Pointer<SAMRAI::hier::PatchHierarchy<NDIM> > patch_hierarchy,
+                               SAMRAI::tbox::Pointer<IBAMR::IBFEMethod> ibfe_method,
+                               const libMesh::Mesh& mesh,
+                               const libMesh::BoundaryMesh& bdry_mesh,
+                               const int part,
+                               const int gcw = 1,
+                               bool use_extracted_bdry_mesh = true);
+
+    /*!
+     * Destructor for this class.
+     */
+    ~FESurfaceDistanceEvaluator();
+
+    /*!
+     * Map the triangles intersecting a particular grid cell.
```

Comment:
```suggestion
     * Map the triangles intersecting a particular Cartesian grid cell.
```
Also, is this specialized to triangular meshes? If not, suggest:
```suggestion
     * Map the elements from the finite element mesh intersecting a particular Cartesian grid cell.
```